cli/command/stack/loader/loader.go

```diff
@@ -104,6 +105,19 @@ func GetConfigDetails(composefiles []string, stdin io.Reader) (composetypes.Conf
 func buildEnvironment(env []string) (map[string]string, error) {
 	result := make(map[string]string, len(env))
 	for _, s := range env {
+		if runtime.GOOS == "windows" && len(s) > 0 {
+			// cmd.exe can have special environment variables which names start with "=".
+			// They are only there for MS-DOS compatibility and we should ignore them.
+			// See TestBuildEnvironment for examples.
+			//
+			// https://ss64.com/nt/syntax-variables.html
+			// https://devblogs.microsoft.com/oldnewthing/20100506-00/?p=14133
+			// https://github.com/docker/cli/issues/4078
+			if s[0] == '=' {
+				continue
+			}
+		}
+
 		k, v, ok := strings.Cut(s, "=")
 		if !ok || k == "" {
 			return result, errors.Errorf("unexpected environment %q", s)
```

Comment:
I would love to see this message changed to "unexpected environment **variable** %q". See [my comment on the bug report](https://github.com/docker/cli/issues/4078#issuecomment-1462258810).

Reply:
Possibly we could use `'%s'` as well; the `%q` is nice as it escapes output, but in some cases that makes the output harder to read (`error something "\"\\\\foo\\\\bar\"` etc), so having it print "as-is" could make that more readable.

Reply:
Done. Not entirely sure about using %s instead of %q though (because if the var starts with `=` which isn't explicitly not allowed, can we really assume it's safe to print without escaping?)

Reply:
I think we should be fine? Or were you thinking of these containing control characters (`\033[1;42m`) that would mess up the terminal?